### PR TITLE
add mattiasgustavsson-libs 20200423

### LINKED ISSUE #1622: [request] mattiasgustavsson/libs
### Package Details
  * Package Name/Version: **???/???**
  * Source code: **https://github.com/mattiasgustavsson/libs**

### Description Of The Library / Tool

Single-file public domain libraries for C/C++

Required by #1544 

I don't know how I can name this recipe !!!

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **mattiasgustavsson-libs/20200514**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Closes #1622

### CODE REVIEW

File: recipes/mattiasgustavsson-libs/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200514":
```
Comment (Croydon) [inline recipes/mattiasgustavsson-libs/all/conandata.yml:2]: Sorry for not more specific in my comment on the issue

I could have provided a link to the wiki: https://github.com/conan-io/conan-center-index/wiki/FAQ#what-version-should-packages-use-for-libraries-without-official-releases

The date should be the date of the commit https://github.com/mattiasgustavsson/libs/commit/3da65d818fd4214d608d62666df32a0fc0428c80
```suggestion
  "20200423":
```

File: recipes/mattiasgustavsson-libs/config.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+versions:
+  "20200514":
```
Comment (Croydon) [inline recipes/mattiasgustavsson-libs/config.yml:2]:
```suggestion
  "20200423":
```

File: recipes/mattiasgustavsson-libs/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    license = "Public Domain"
+    topics = ("utilities", "mattiasgustavsson", "libs")
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        print(self.conan_data)
```
Comment (Croydon) [inline recipes/mattiasgustavsson-libs/all/conanfile.py:21]:
```suggestion
```

File: recipes/mattiasgustavsson-libs/all/test_package/main.cpp [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+int main() {
+
+    rnd_pcg_t pcg;
+	rnd_pcg_seed( &pcg, 0u ); // initialize generator
+
+	std::cout << rnd_pcg_next( &pcg ) << std::endl;
+
+    return 0;
+}
```
Comment (Croydon) [inline recipes/mattiasgustavsson-libs/all/test_package/main.cpp:15]: Fix indentations

File: recipes/mattiasgustavsson-libs/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    description = "Single-file public domain libraries for C/C++"
+    homepage = "https://github.com/mattiasgustavsson/libs"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = ("Public domain", "MIT")
+    topics = ("utilities", "mattiasgustavsson", "libs")
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
```
Comment (uilianries) [inline recipes/mattiasgustavsson-libs/all/conanfile.py:18]:
```suggestion
        return os.path.join(self.source_folder, "source_subfolder")
```

File: recipes/mattiasgustavsson-libs/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    def package(self):
+        self.copy(pattern="*.h", dst="include", src=self._source_subfolder)
+
+        with open(os.path.join(self.source_folder, self._source_subfolder, "thread.h")) as f:
+            content_lines = f.readlines()
+        license_content = []
+        for i in range(1449, 1498):
+            license_content.append(content_lines[i][:-1])
+
+        tools.save(os.path.join(self.package_folder, "licenses", "license.txt"), "\n".join(license_content))
```
Comment (uilianries) [inline recipes/mattiasgustavsson-libs/all/conanfile.py:34]:
```suggestion
    def _extract_licenses(self):
        header = tools.load(os.path.join(self._source_subfolder, "thread.h"))
        mit_content = header[header.find("ALTERNATIVE A - "):header.find("ALTERNATIVE B -")]
        tools.save("LICENSE_MIT", mit_content)
        unlicense_content = header[header.find("ALTERNATIVE B - "):header.rfind("*/", 1)]
        tools.save("LICENSE_UNLICENSE", unlicense_content)

    def package(self):
        self.copy(pattern="*.h", dst="include", src=self._source_subfolder)
        self._extract_licenses()
        self.copy("LICENSE_MIT", dst="licenses")
        self.copy("LICENSE_UNLICENSE", dst="licenses")
```

**Pull request merged**